httpgrpc/protocol_versions.go

```diff
@@ -30,3 +36,24 @@ const (
 	UnaryRpcContentType_V1  = "application/x-protobuf"
 	StreamRpcContentType_V1 = "application/x-httpgrpc-proto+v1"
 )
+
+func getCodec(contentType string) encoding.Codec {
```

Comment:
For robustness, since content type can include additional options, like charset, I think you want to use `mime.ParseMediaType` and then just use the resulting media type. And maybe also disallow any explicit charset that is not "utf8", too (or a TODO for that).